src/parcsr_mv/par_csr_matop.c

```diff
@@ -483,9 +483,20 @@ hypre_ParMatmul( hypre_ParCSRMatrix  *A,
    if (num_rownnz_diag_A != num_rows_diag_A &&
        num_rownnz_offd_A != num_rows_offd_A )
    {
-      hypre_MergeOrderedArrays(num_rownnz_diag_A, A_diag_ir,
-                               num_rownnz_offd_A, A_offd_ir,
-                               &num_rownnz_A, &rownnz_A);
+      hypre_IntArray arr_diag;
+      hypre_IntArray arr_offd;
+      hypre_IntArray arr_rownnz;
+
+      hypre_IntArrayData(&arr_diag) = A_diag_ir;
+      hypre_IntArrayData(&arr_offd) = A_offd_ir;
+      hypre_IntArraySize(&arr_diag) = num_rownnz_diag_A;
+      hypre_IntArraySize(&arr_offd) = num_rownnz_offd_A;
+      hypre_IntArrayMemoryLocation(&arr_rownnz) = memory_location_A;
+
+      hypre_MergeOrderedArrays(&arr_diag, &arr_offd, &arr_rownnz);
```

Comment:
I'm wondering if we should change this function name to `hypre_IntArrayMergeOrdered` since it operates on a `hypre_IntArray` now. This is very minor...

Reply:
Fixed. thanks!